### DISCUSSION TITLE
How to find any element by its text, without specifying its type

### DISCUSSION THREAD

**Discussion by igorgiumellizup** [cypress-io/cypress#17287] [Questions and Help]
Hello.

I have the following code that tries to find any element by its text value (inner html value):

```
   getAnyElementByExactText(text: string) 
   {
        cy.get('html').children().should((elem) => {
            expect(elem.text().trim()).to.equal(text);
        });
    }
```
But the part _cy.get('html').children()_ is not working properly, as it is returning undefined.

When I specify the type, it works:

```
   getAnyElementByExactText(text: string) 
   {
        cy.contains('button', text).should((elem) => {
            expect(elem.text().trim()).to.equal(text);
        });
    }
```

**Comment by bahmutov** [comment]
Check out the cy.contains command

**Comment by igorgiumellizup** [comment]
I'm using contains as follows:

```
    static getElementByExactText(text: string) {
        return cy.contains(text).should($elem => {
            // @ts-ignore
            expect($elem.text().trim()).to.be.eq(text)
        });
    }
```

Now, image that I have a web page with two elements:

```
ContextTwo
Context
```

I want to get the element with text "Context". **The problem is**, it is getting the first element and failing in the code expect_($elem.text().trim()).to.be.eq(text)_, **instead of trying to continue the searching**.

**Comment by bahmutov** [comment]
It finds the first element even if you pass “ContextTwo”?

**Reply by igorgiumellizup** [reply]
I edited my previous post

**Reply by bahmutov** [reply]
You can use regular expression to match the text precisely, see https://docs.cypress.io/api/commands/contains#Regular-Expression

**Reply by igorgiumellizup** [reply]
Thanks, I managed to get the element with the following code:

```
static getElementByExactText(elementTag: string, text: string) {
        return cy.contains(elementTag, new RegExp("^\\s*" + text + "\\s*$"))
    }
```
